Handle URLs in meta/main.yml dependencies

willthames opened the pull request:
e.g

```
dependencies:
- git+https://github.com/blah/blah.git,v1.2
```

```
$ ansible-playbook playbook.yml
Using /home/wthames/src/ansible-playbooks/ansible.cfg as config file
ERROR! the role 'git+https://github.com/blah/blah.git,v1.2' was not found in /home/wthames/src/ansible-playbooks/roles:./roles:...
```

Use `RoleRequirement` rather than `RoleDefinition` for handling role inclusions

Handle of comma separated specified versions

Allow old style string role specifications (no need for `{ role: rolespecification }` when `rolespecification` will suffice.

This is for backward compatibility with 1.8 style rolesfiles and meta/main.yml files

mattclay commented:
All CI jobs except the "other" test have failed: https://app.shippable.com/runs/58925ae99e41480f00eea544/summary

willthames committed:
Handle URLs in meta/main.yml dependencies

e.g
```
dependencies:
- git+https://github.com/blah/blah.git,v1.2
```

```
$ ansible-playbook playbook.yml
Using /home/wthames/src/ansible-playbooks/ansible.cfg as config file
ERROR! the role 'git+https://github.com/blah/blah.git,v1.2' was not found in /home/wthames/src/ansible-playbooks/roles:./roles:...
```

Use RoleRequirement rather than RoleDefinition for handling role
inclusions

Handle of comma separated specified versions

Allow old style string role specifications (no need for
`{ role: rolespecification }` when `rolespecification` will
suffice.

This is for backward compatibility with 1.8 style rolesfiles and
meta/main.yml files

mattclay commented:
CI failure due to a PEP 8 issue:

```
2017-02-02 06:27:12 ERROR: PEP 8: lib/ansible/playbook/role/requirement.py:110:5: E129 visually indented line with same indent as next logical line (current)
```

There are also unit test failures:

```
2017-02-02 06:29:51 E           AnsibleParserError: A malformed list of role dependencies was encountered.
2017-02-02 06:29:51 E           
2017-02-02 06:29:51 E           The error appears to have been in '/etc/ansible/roles/role1/meta/main.yml': line 2, column 13, but may
2017-02-02 06:29:51 E           be elsewhere in the file depending on the exact syntax problem.
2017-02-02 06:29:51 E           
2017-02-02 06:29:51 E           (could not open file to display line)
2017-02-02 06:29:51 
2017-02-02 06:29:51 lib/ansible/playbook/role/metadata.py:93: AnsibleParserError
```

All integration tests are also failing with a similar error.

willthames commented:
Closing, I don't care about this that much.
